src/gromacs/nblib/CMakeLists.txt

```diff
@@ -38,13 +38,47 @@
 # \author Sebastian Keller <keller@cscs.ch>
 #
 
-file(GLOB NBLIB_SOURCES *.cpp)
+add_library(nblib "")
 
-install(FILES
+target_sources(nblib
+        INTERFACE
+        bondtypes.h
```

Comment:
Would your suggestion then be to move these into the `add_library()` call? Any idea why there has been pushback on this?

Reply:
Whether files are listed in `add_library` or `target_sources` is not the issue.

Theoretically, you should never have to list header files as "source" files for CMake. They should be discoverable through one of the INCLUDE_DIRECTORIES, so they will be discovered and checked by the build system. If you choose to list headers as "source" files, CMake marks them as special and treats them sensibly.
* Option 1: don't list them at all.
* Option 2: if you want different headers associated with different targets, put them in separate directories and make sure that each directory is only exposed in the INTERFACE_INCLUDE_DIRECTORIES of a particular target.
* Option 3: Articulate the reason that you want header files to be part of INTERFACE_SOURCES and be ready for push-back.

I don't know if the objection was related more to the fact that CMake calls files "sources" or to the fact that it is arguably heavy-handed (since CMake doesn't need the headers to be listed). If you browse around online a bit, you'll find some people vehemently decry the practice, some of whom go on to post follow-ups that it doesn't seem that terrible after all.

I think it makes sense for the headers to be in a target property somewhere, and INTERFACE_SOURCES seems like as good a place as any. Plus, it adds a little bit more robustness to regenerate the build system when headers are added, removed, or converted to or from generated files. It also helps CMake-aware development environments to correctly understand dependencies.